### ISSUE TITLE
Azure deployment fails with BitBucket with an OAuthException

### ISSUE THREAD

**Issue by adaam2** [projectkudu/kudu#1885]
The deployment via source control option in the Azure portal usually works very well, but I'm in the process of setting up an old static site as an Azure web app and I'm experiencing an issue with deployment using Kudu.  I've added my solution into source control on BitBucket but I receive an Exception with this Stack trace:

```
Failed to set up continuous deployment for web app [NAME OF APP], using Bitbucket. Repository 'UpdateSiteSourceControl' operation failed with Microsoft.Web.Hosting.SourceControls.OAuthException: Bitbucket AddSSHKey: (500) InternalServerError.
at Microsoft.Web.Hosting.SourceControls.BitbucketV2Proxy.<AddSSHKey>d__57.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.Web.Hosting.Administration.BitbucketV2SiteRepositoryProvider.<UpdateSiteSourceControl>d__6.MoveNext().

```
I am unsure of what is going wrong here, as I have added my SSH key to my BitBucket account and the deployment keys inside of the Project on BitBucket seem to be setup correctly. What does this error indicate?

n.b. I also tried to setup deployment manually as per the instructions at https://github.com/projectkudu/kudu/wiki/Continuous-deployment but `curl`ing the SCM url (e.g. https://$mysite:BigRandomPassword@mysite.scm.azurewebsites.net/api/sshkey?ensurePublicKey=1) returned a `401 - Unauthorized: Access is denied due to invalid credentials.` error.

Thanks,

Adam

**Comment by davidebbo** [comment]
Similar issue also just came up [on forum](https://social.msdn.microsoft.com/Forums/en-US/1742d905-982d-4a25-a596-91fe4e19b3ed/web-app-continuous-deployment-oauthexception-addsshkey-500?forum=windowsazurewebsitespreview). Maybe Bitbucket was having some temporary issue. Does it still repro for you? If so, please see questions in forum. Thanks!

**Comment by adaam2** [comment]
I'll check it out tomorrow to see if it was just a temporary issue!

**Comment by badescuga** [comment]
@adaam2 

Is the issue still persisting? Could you try using the old portal? 

![Link to BitBucket](https://cloud.githubusercontent.com/assets/11161526/12945101/f662a2ae-cff3-11e5-9346-9c6660ed14ce.png)

**Comment by adaam2** [comment]
Hi @badescuga 

We were experiencing lots of issues yesterday evening and earlier this morning with deployments from Source Control from BitBucket. We found that deployments weren't syncing and also failing when manually synced across using both the new portal and the old one. As a stop gap, I managed to get deployments working by setting up a local Git repository on the Web App and triggering the web hook SCM URL for the Web App to trigger a build.

However, things seem to be working fine now. We also experienced a major bug with one of our sites that sat on one of our main App service plans inside of Azure which we believe had a knock on effect on all other sites (oh and the Azure interface seemed to be flat out refusing to play ball this morning as well)

So in conclusion I am none the wiser as to what was happening here. Never experienced anything like this before with Kudu!

**Comment by adaam2** [comment]
n.b. Closing as seems to be OK now.

**Comment by davidebbo** [comment]
My guess is that it was a bitbucket outage. Looking at http://status.bitbucket.org/#week, it looks like they had about 90 minutes of downtime yesterday. Not sure if the time matches what you saw.

**Comment by adaam2** [comment]
Looks like it! Thanks for your help.

**Comment by WernerMairl** [comment]
May be we have the same at this moment 
Anyone can confirm ? (BB-Status is green) ?
